src/main/java/com/echostreams/pulsar/jms/jndi/JNDIObjectFactory.java

```diff
@@ -0,0 +1,148 @@
+     * the caller to select one.  A good rule of thumb is to select the
+     * "deepest" context available.
+     * If <code>nameCtx</code> is null, <code>name</code> is relative
+     * to the default initial context.  If no name is being specified, the
+     * <code>name</code> parameter should be null.
+     * If a factory uses <code>nameCtx</code> it should synchronize its use
+     * against concurrent access, since context implementations are not
+     * guaranteed to be thread-safe.
+     * <p>
+     *
+     * @param obj         The possibly null object containing location or reference
+     *                    information that can be used in creating an object.
+     * @param name        The name of this object relative to <code>nameCtx</code>,
+     *                    or null if no name is specified.
+     * @param nameCtx     The context relative to which the <code>name</code>
+     *                    parameter is specified, or null if <code>name</code> is
+     *                    relative to the default initial context.
+     * @param environment The possibly null environment that is used in
+     *                    creating the object.
+     * @return The object created; null if an object cannot be created.
+     * @throws Exception if this object factory encountered an exception
+     *                   while attempting to create an object, and no other object factories are
+     *                   to be tried.
+     * @see NamingManager#getObjectInstance
+     * @see NamingManager#getURLContext
+     */
+    @Override
+    public Object getObjectInstance(Object obj, Name name, Context nameCtx, Hashtable<?, ?> environment) throws Exception {
+        Object result = null;
+
+        if (obj instanceof Reference) {
+            Reference ref = (Reference) obj;
+            String clazz = ref.getClassName();
+
+            // ConnectionFactory or QueueConnectionFactory or TopicConnectionFactory
+            if (clazz.equals(PulsarConnectionFactory.class.getName())) {
+
+                StringRefAddr serverClass =
+                        (StringRefAddr) ref.get("serverClass");
+                String serverClassName = (String) serverClass.getContent();
+
+                // get a list of string properties and store them in a
+                // hashtable
+                HashMap properties = new HashMap();
+                Enumeration iter = ref.getAll();
+                while (iter.hasMoreElements()) {
+                    StringRefAddr addr = (StringRefAddr) iter.nextElement();
+                    properties.put(addr.getType(), addr.getContent());
+                }
+
+                // create the factory
+                //result = new PulsarConnectionFactory(serverClassName, properties, environment);
+            }
+
+        } else {
+            throw new IllegalStateException("Object is not a reference : " + obj);
+        }
+
+        return result;
```

Comment:
This will always be null. Are you not supposed to create the Factory?

Reply:
Sorry, forget to refactor this. This class is redundant. Actually class JNDIReferenceFactory is being used.